docs/contributing/999-rfcs/2024-06-02-ts-client.md

```diff
@@ -0,0 +1,354 @@
+    b.put("key", "value");
+  }
+}
+```
+
+With some changes, with syntax could also be used as a way to create a preflight container for lifted data.
+
+```wing
+let b = new cloud.Bucket();
+let lifts = @lift({ b: [put] });
+//  ^^^^^ is a std.LiftQualificationsContainer
+```
+
+This is very similar to the `lift` mechanism in `@wingcloud/framework`
+
+```ts
+const b = new cloud.Bucket(app, "Bucket");
+const lifts = lift({ bucket }).grant({ bucket: ["put"] });
+```
+
+In the [RFC for explicit lift qualifications](https://github.com/winglang/wing/blob/main/docs/contributing/999-rfcs/2024-03-14-explicit-lift-qualification.md), there is a mention of a `std.LiftQualificationsContainer`.
+
+If this data structure is augmented to also include type information, it could be used to generate the client during preflight.
+The resulting data would look very similar to the existing _liftMap, with the addition of the link to the `.d.ts` generated by the compiler for the aggregate type information.
+
+### Client Contents
+
+The client package will contain the following:
+
+- package.json
+
+Minimal package.json with the name and version. For MVP, the package will not have any dependencies because the client code is self-contained.
+
+- index.js
+
+Exports the `Client` class. In this class constructor, the context is read from the environment variable `WING_CONTEXT` (or constructor arg) and is used to set the environment variables for the current process. Each lifted object will then be instantiated which utilizes the existing `_toInflight` logic and will consume all the necessary environment variables.
+
+(Future Improvements: The lifted clients could be lazily instantiated?)
+
+- index.d.ts
+
+Exports the `Client` class, with fields for each lifted object. Each field will reference a type from `lifts.d.ts`.
+
+- lifts.d.ts
+
+The resulting types from `@lift`
+
+## Example Usage
+
+### EKS
+
+This walks through the setup of a client by a central platform team for an EKS cluster in AWS.
+
+> **Note:**
+> In this example, wing is also used to manage some of the kubernetes manifests. This is not a requirement, but greatly shortens the example
+> and is a recommended setup. If not, imagine all the data from wing was simply exported (e.g. `cdktf.Output`) and managed in kubernetes manifests manually.
+
+We start by creating the client and lifting a bucket:
+
+```wing
+bring cloud;
+
+let bucket = new cloud.Bucket();
+
+let client = new typescript.Client(
+  name: "@acme/infra-client",
+  lifts: @lift({ bucket: [put] }),
+);
+```
+
+To extract the necessary permissions, we set up the wing.toml file to create a new policy:
+
+```toml
+[tf-aws]
+client_policy = "AcmeClientPolicy"
+```
+
+After this policy is created, a role and associated service account must be created for application pods to use it.
+This is only needed for `tf-aws` deployment so it can either but inside a check to the `WING_TARGET` environment variable or in a separate class/file:
```

Comment:
```suggestion
This is only needed for `tf-aws` deployment so it can either be inside a check to the `WING_TARGET` environment variable or in a separate class/file:
```